src/library/autodj/autodjprocessor.cpp

```diff
@@ -142,6 +144,13 @@ AutoDJProcessor::~AutoDJProcessor() {
     delete m_pAutoDJTableModel;
 }
 
+void AutoDJProcessor::registerSelectors() {
+    LinearSelector* ls = new LinearSelector(this);
```

Comment:
you should avoid plain new. you can use std::make_unique here and std::move the object to the registerSelector(ls);
or maybe more elegant use a template function to register, something like 
registerSelector<RandomSelector>();

Reply:
if you store the pointer in a qHash you cannot use unque pointers. Use std::hash or QSharedPointer.